tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_files/invalid_use_of_overload_acceptable.py

```diff
@@ -0,0 +1,32 @@
+# Test file for InvalidUseOfOverload checker
+
+from typing import Awaitable, overload, Union
+
+class testingOverload:
+    @overload
+    def double(a: str)  -> Awaitable[int]:
+        ...
+
+    @overload
+    def double(a: int) -> Awaitable[int]:
+        ...
+
+    async def double(a: Union[str, int]) -> int:
```

Comment:
after further discussion could we remove this scenario from the checker for now, unless you have a use case for this -- it does not seem to appear in our repo

Reply:
I can' remove this out, It was just to make sure that all scenarios were checked.